app/screens/sso/sso_with_webview.tsx

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {Text, View} from 'react-native';
+import CookieManager from 'react-native-cookies';
+import {SafeAreaView} from 'react-native-safe-area-context';
+import {WebView} from 'react-native-webview';
+import {WebViewErrorEvent, WebViewMessageEvent, WebViewNavigation, WebViewNavigationEvent} from 'react-native-webview/lib/WebViewTypes';
+import urlParse from 'url-parse';
+
+import {Client4} from '@mm-redux/client';
+import type {Theme} from '@mm-redux/types/preferences';
+
+import Loading from 'app/components/loading';
+import StatusBar from 'app/components/status_bar';
+import {ViewTypes} from 'app/constants';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+const HEADERS = {
+    'X-Mobile-App': 'mattermost',
+};
+
+const postMessageJS = "window.postMessage(document.body.innerText, '*');";
+
+// Used to make sure that OneLogin forms scale appropriately on both platforms.
+const oneLoginFormScalingJS = `
```

Comment:
@enahum Is this okay? Never seen this else where in our codebase. I mean having code like this injected into webview. Just want make sure you're okay with this.

Reply:
This code already existed and it is not ok it is absolutely needed

Reply:
Gotcha. Thanks for clarifying